File: 203_RemoveLinkedListElement.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        # # v1  iterative
        # dummyhead=ListNode(0)
        # dummyhead.next=head
        # rethead=dummyhead
        # while dummyhead.next: 
        #     if dummyhead.next.val==val:
        #         dummyhead.next=dummyhead.next.next
        #     else:
        #         dummyhead=dummyhead.next
        # return rethead.next
        
        # v2 recursive
        if head==None:
            return None
        if head.val==val:
            return self.removeElements(head.next,val)
        head.next=self.removeElements(head.next,val)
        return head

File: LeetCode/203_RemoveLinkedListElement.go
// https://leetcode.com/problems/remove-linked-list-elements/
// 2022年07月24日 22:22:21

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func removeElements(head *ListNode, val int) *ListNode {
    // v1 ugly
    for head!=nil && head.Val==val{
        head=head.Next
    }
    if head==nil || ( head.Next==nil && head.Val!=val) {
        return head
    }else if head.Next==nil && head.Val==val{
        return nil
    }
    curr:=head
    var ret *ListNode
    ret=head
    for curr!=nil && curr.Next!=nil{
        if curr.Next.Val == val{
            curr.Next=curr.Next.Next
        }else{  
            curr=curr.Next
        }
    }
     
    return ret
}